### ISSUE TITLE
Make generic device class available to users in client-superstaq

### ISSUE THREAD

**Issue by ColinCampbellCQ** [Infleqtion/client-superstaq#1077]
### What is happening?

I am trying to determine the impact of different device architectures (coupling maps, gate sets, etc.) on resources of compiled circuits. Initially, I am hoping to see how different coupling maps result in more/less CZ gates on the cq_sqorpius backend, but later, I would like to see the impact of different gate sets. I think a more abstract Device being available in the client would help with this process.

### What are the current alternatives?

I can utilize functionality on the server side as an alternative, but I would have to copy code/data from the environment where I am doing my experiments into the environment where I develop server-superstaq.

### Any additional context?

There is a discussion to be had about what reasonable limits should be placed on an abstract architecture being made available client side. How exotic could I make my gates and still expect the highest quality of performance? How detailed should I be able to make an abstract architecture? Even without having the answers to these questions, I think that something should be available for users to experiment with different design components. What do you think?

**Comment by cdbf1** [comment]
Yeah, I think as much as anything this is a question about how much we open source.
As a work-around, could you just install the server-side `superstaq` package into the environment you are using to experiment in? This has been my approach to utilize server-side functionality in the past.

**Comment by ColinCampbellCQ** [comment]
Is that simple to do? That sounds a lot better than what I was planning to do, but I am not sure how to do it.

**Comment by cdbf1** [comment]
> Is that simple to do? That sounds a lot better than what I was planning to do, but I am not sure how to do it.

Sure, I just do 
```
pip install [-e if you want] <relative path to a local copy of server_superstaq>
```

**Comment by vtomole** [comment]
@ColinCampbellCQ You can passing in `coupling_map=[(0,1), (0,3)]`

**Comment by cdbf1** [comment]
Closing as fixed
